docs/python/programming-guide.md

```diff
@@ -7,321 +7,177 @@ pagination_prev: quickstart
 
 ```
-14:54:13 [    INFO] (dbos:main.py:39) >>> STEP 1: Signed the Guestbook: {"ip_address": "...", "greeted_name": "Mike", "greeted_ts": "2024-08-29T21:54:13.832Z"}
-14:54:13 [    INFO] (dbos:main.py:47) >>> STEP 2: Greeting to Mike recorded in the database!
+14:54:13 [    INFO] (dbos:main.py:21) >>> STEP 1: Signed the guestbook for Mike
+14:54:13 [    INFO] (dbos:main.py:28) >>> STEP 2: Greeting to Mike recorded in the database!
 ```
 
-You can visit the URL `https://demo-guestbook.cloud.dbos.dev/greetings/your-key-value` to see all the Guestbook greetings made with your key. Old greetings and keys are removed after a few days.
-
-## 5. Durable Execution with Workflows
+## 3. Durable Execution with Workflows
 
 Next, we want to **durably execute** our application: guarantee that it inserts exactly one database record per guestbook signature, even if interrupted or restarted.
-DBOS makes this easy with [workflows](../python/tutorials/workflow-tutorial.md).
-To add one, change your `main.py` like so:
+DBOS makes this easy with [workflows](./tutorials/workflow-tutorial.md).
+Copy the following code into your `main.py`:
 
 ```python
-import json
-import os
-
 import requests
 from dbos import DBOS
 from fastapi import FastAPI
-from fastapi.responses import HTMLResponse
-from .schema import greetings
+
+from .schema import dbos_hello
 
 app = FastAPI()
 DBOS(fastapi=app)
 
-# Omitted for brevity: def get_greetings()
-
 @DBOS.step()
 def sign_guestbook(name: str):
-    key = os.environ.get("GUESTBOOK_KEY", None)
-    if key is None or len(key) != 36:
-        raise Exception("Please set the guestbook key in dbos-config.yaml")
-
-    url = "https://demo-guestbook.cloud.dbos.dev/record_greeting"
-    headers = {"Content-Type": "application/json"}
-    payload = {"key": key, "name": name}
-
-    response = requests.post(url, headers=headers, json=payload)
-    response_str = json.dumps(response.json())
-    if not response.ok:
-        raise Exception(f"Error signing guestbook: {response_str}")
-
-    DBOS.logger.info(f">>> STEP 1: Signed the Guestbook: {response_str}")
-
+    response = requests.post(
+        "https://demo-guest-book.cloud.dbos.dev/record_greeting",
+        headers={"Content-Type": "application/json"},
+        json={"name": name},
+    )
+    DBOS.logger.info(f">>> STEP 1: Signed the guestbook for {name}")
 
 @DBOS.transaction()
-def insert_greeting(name: str, note: str):
-    DBOS.sql_session.execute(greetings.insert().values(name=name, note=note))
+def insert_greeting(name: str) -> str:
+    query = dbos_hello.insert().values(name=name)
+    DBOS.sql_session.execute(query)
     DBOS.logger.info(f">>> STEP 2: Greeting to {name} recorded in the database!")
 
-
+@app.get("/greeting/{name}")
 @DBOS.workflow()
-def greeting_workflow(name: str, note: str):
+def greeting_endpoint(name: str):
```

Comment:
Reminder just in case: there's a problem here that caused us to use `start_workflow`. On Chrome, the browser would try to reconnect after ctrl+c which resulted in two workflows and confusing output.